src/Optimizer/Transformer/MinifyHtml.php

```diff
@@ -98,12 +106,20 @@ private function minifyNode(DOMNode $node, ErrorCollection $errors, $canCollapse
         }
 
         if ($node->hasChildNodes()) {
+            if ($this->isMustacheTemplate($node)) {
+                $this->isInsideMustacheTemplate = true;
+            }
+
             foreach ($node->childNodes as $childNode) {
                 $nodesToRemove = array_merge(
                     $nodesToRemove,
                     $this->minifyNode($childNode, $errors, $canCollapseWhitespace, $inBody)
                 );
             }
+
+            if ($this->isMustacheTemplate($node)) {
```

Comment:
Very minor point. The check could be stored in a variable to avoid running it twice. So like:

```php
$isNodeMustacheTemplate = $this->isMustacheTemplate($node);
if ($isNodeMustacheTemplate) {
    $this->isInsideMustacheTemplate = true;
}
// ...
if ($isNodeMustacheTemplate) {
    $this->isInsideMustacheTemplate = false;
}
```

Reply:
Also, should this be checking if `$this->isInsideMustacheTemplate` is not already true? While I don't think nested `template` elements are a thing right now, it might make sense to harden this by switching `isInsideMustacheTemplate` from a boolean to an integer depth count. 

So then you'd do:

```php
$isNodeMustacheTemplate = $this->isMustacheTemplate($node);
if ($isNodeMustacheTemplate) {
    $this->mustacheTemplateDepth++;
}
// ...
if ($isNodeMustacheTemplate) {
    $this->mustacheTemplateDepth--;
}
```

And then instead of checking `$this->isInsideMustacheTemplate` below, you'd check `$this->mustacheTemplateDepth > 0` (or just use its truthy value as well).

Reply:
Hi @westonruter thanks for the suggestions. I've raised [a small PR](https://github.com/ampproject/amp-toolbox-php/pull/309) including your suggesting code.